src/conspiracies/prompt_relation_extraction/template_class.py

```diff
@@ -0,0 +1,579 @@
+    such as GPT-3.
+    """
+
+    def __init__(
+        self,
+        task_description: Optional[str],
+        examples: Optional[Tuple[List[Doc], List[List[SpanTriplet]]]] = None,
+    ):
+        """
+        Args:
+            task_description: A description of the task
+            examples: A tuple of a spacy Docs and a corresponding list of their
+                triplets.
+        """
+        if examples:
+            self.examples = examples
+        else:
+            self.examples = load_gold_triplets()
+        self.task_description = task_description
+
+    @abstractmethod
+    def create_prompt(self, target: str) -> str:
+        """Create a prompt based on the task description and examples and
+        target."""
+        pass
+
+    @abstractmethod
+    def parse_prompt(self, prompt: str) -> List[StringTriplet]:
+        """Parse a prompt into a target tweet and triplets."""
+        pass
+
+    def set_examples(self, examples: Tuple[List[Doc], List[List[SpanTriplet]]]):
+        """Updates examples."""
+        self.examples = examples
+
+
+class PromptTemplate1(PromptTemplate):
+    def create_prompt(
+        self,
+        target: str = "",
+    ) -> str:
+        """Create a prompt template on the form
+
+        ```
+        {task_description}
+
+        Tweet: {tweet1}
+        Triplet: {triplets1}
+        ...
+        Tweet: {tweetN}
+        Triplet: {tripletsN}
+
+        Tweet: {target_tweet}
+        ```
+
+        Args:
+            target: The tweet to be annotated.
+        """
+        examples_str = "---\n\n"
+        for tweet, triplets in zip(*self.examples):
+            examples_str += f"Tweet: {tweet.text}\n"
+
+            if len(triplets) == 0:
+                examples_str += "Triplets: Null \n\n"
+            else:
+                for i, triplet in enumerate(triplets):
+                    triplet_str = "({subj}) ({pred}) ({obj})"
+                    triplet_str = triplet_str.format(
+                        subj=triplet.subject.text,
+                        pred=triplet.predicate.text,
+                        obj=triplet.object.text,
+                    )
+                    if i == 0:
+                        examples_str += f"Triplets: {triplet_str}\n"
+                    else:
+                        examples_str += f"\t{triplet_str}\n"
+
+            examples_str += "---\n"
+            tweet_string = f"{self.task_description}\n\n{examples_str}\nTweet:"
+            prompt = tweet_string
+
+        if target:
+            return prompt + f"{target}\nTriplets:"
+        return prompt
+
+    @staticmethod
+    def create_string_triplets(tweet, triplet):
+        subject, predicate, object = triplet
+        return StringTriplet(
+            subject=subject,
+            predicate=predicate,
+            object=object,
+            text=tweet,
+        )
+
+    def parse_prompt(self, prompt_response: str, target_tweet: str) -> List[StringTriplet]:  # type: ignore
```

Comment:
```suggestion
    def parse_prompt(self, prompt_response: str, target_tweet: str) -> List[StringTriplet]:
```

Reply:
@stinenyhus: It is me who added the type ignore. It simply avoid says to mypy that it should not check the type.

(before the function was completed it complained about the return not being the same as specified in the type hint.)